### ISSUE TITLE
Crash when activating an account

### ISSUE THREAD

**Issue by Nihim** [thunderbird/thunderbird-android#2708]
I updated some weeks ago to a new phone with android nougat (7.0). I 've read up that syncing is not working properly on 7.0 and till a fix is found I can deal with doing manual upgrades.

But I am experiencing another bug. I have 6 email accounts added (imported from a phone with android 6.0). First one on the list works as expected. For the rest if I press on them (A), K-9 closes. Long pressing shows the pop up menu but has Activate (2nd screenshot). Pressing Activate closes K-9.

If I press on the directory icon on the right (B) on one of the not working emails, it enters in the Folders menu and I can access all directories, settings and sync the account just fine.

So firing up K-9 only the 1st account on the top works normally. The rest if I press on them crash K-9 and only work fine if I enter in the Folders menu.

![screenshot_20170827-115206](https://user-images.githubusercontent.com/8713434/29748877-45e2cd72-8b28-11e7-8a22-0af6db769141.png)

![screenshot_20170827-115218](https://user-images.githubusercontent.com/8713434/29748891-955d4b0c-8b28-11e7-9821-45587954c0b5.png)

### Expected behavior
It should be able to enter in the accounts without a force-close

### Actual behavior
K-9 closes.

### Steps to reproduce
1. Open K-9
2. Click on any account except the first.
3. K-9 closes

### Environment
K-9 Mail version: 5.207

Android version: 7.0

Account type (IMAP, POP3, WebDAV/Exchange): imap

**Comment by cketti** [comment]
Please provide a debug log. See https://github.com/k9mail/k-9/wiki/LoggingErrors

**Comment by Nihim** [comment]
There you go, I think and hope I 've removed any private information from it.

[k9-log.txt](https://github.com/k9mail/k-9/files/1258171/k9-log.txt)

**Comment by cketti** [comment]
Thanks.

Stack trace from the log:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'android.text.Editable android.widget.EditText.getText()' on a null object reference
	at com.fsck.k9.activity.Accounts$PasswordPromptDialog.show(Accounts.java:894)
	at com.fsck.k9.activity.Accounts$PasswordPromptDialog.show(Accounts.java:750)
	at com.fsck.k9.activity.Accounts.promptForServerPasswords(Accounts.java:687)
	at com.fsck.k9.activity.Accounts.onActivateAccount(Accounts.java:672)
	at com.fsck.k9.activity.Accounts.onOpenAccount(Accounts.java:648)
	at com.fsck.k9.activity.Accounts.onItemClick(Accounts.java:1220)
	at android.widget.AdapterView.performItemClick(AdapterView.java:313)
	at android.widget.AbsListView.performItemClick(AbsListView.java:1201)
	at android.widget.AbsListView$PerformClick.run(AbsListView.java:3195)
	at android.widget.AbsListView$3.run(AbsListView.java:4138)
	at android.os.Handler.handleCallback(Handler.java:761)
	at android.os.Handler.dispatchMessage(Handler.java:98)
	at android.os.Looper.loop(Looper.java:156)
	at android.app.ActivityThread.main(ActivityThread.java:6523)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:941)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:831)
```

**Comment by cketti** [comment]
@Nihim: Have the non-working accounts be configured to use client certificate authentication?

**Comment by Nihim** [comment]
All the accounts use the same configuration and it is simple password authentication, but if you want to point me to the exact setting I can verify that for you.

**Comment by Nihim** [comment]
bump

@cketti got this in the internal mail 1 hour ago:

>

>  K9-Mail version: 5.207
> Device make: HUAWEI
> Device model: FRD-L09
> Android version: 7.0
> 
> com.fsck.k9.mail.MessagingException: IO Error
> at com.fsck.k9.mail.store.imap.ImapFolder.ioExceptionHandler(ImapFolder.java:1366)
> at com.fsck.k9.mail.store.imap.ImapFolder.copyMessages(ImapFolder.java:378)
> at com.fsck.k9.mail.store.imap.ImapFolder.moveMessages(ImapFolder.java:388)
> at com.fsck.k9.mail.store.imap.ImapFolder.delete(ImapFolder.java:422)
> at com.fsck.k9.controller.MessagingController.processPendingMoveOrCopy(MessagingController.java:2072)
> at com.fsck.k9.controller.MessagingController.processPendingCommandsSynchronous(MessagingController.java:1752)
> at com.fsck.k9.controller.MessagingController.access$800(MessagingController.java:115)
> at com.fsck.k9.controller.MessagingController$13.run(MessagingController.java:1690)
> at com.fsck.k9.controller.MessagingController.runInBackground(MessagingController.java:204)
> at com.fsck.k9.controller.MessagingController.access$000(MessagingController.java:115)
> at com.fsck.k9.controller.MessagingController$1.run(MessagingController.java:173)
> at java.lang.Thread.run(Thread.java:776)
> Caused by: java.net.SocketTimeoutException: Read timed out
> at com.android.org.conscrypt.NativeCrypto.SSL_read(Native Method)
> at com.android.org.conscrypt.OpenSSLSocketImpl$SSLInputStream.read(OpenSSLSocketImpl.java:758)
> at java.util.zip.InflaterInputStream.fill(InflaterInputStream.java:243)
> at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:158)
> at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
> at java.io.BufferedInputStream.read(BufferedInputStream.java:254)
> at com.fsck.k9.mail.filter.PeekableInputStream.peek(PeekableInputStream.java:36)
> at com.fsck.k9.mail.store.imap.ImapResponseParser.readResponse(ImapResponseParser.java:38)
> at com.fsck.k9.mail.store.imap.ImapResponseParser.readResponse(ImapResponseParser.java:30)
> at com.fsck.k9.mail.store.imap.ImapResponseParser.readStatusResponse(ImapResponseParser.java:88)
> at com.fsck.k9.mail.store.imap.ImapConnection.executeSimpleCommand(ImapConnection.java:634)
> at com.fsck.k9.mail.store.imap.ImapConnection.executeSimpleCommand(ImapConnection.java:620)
> at com.fsck.k9.mail.store.imap.ImapFolder.executeSimpleCommand(ImapFolder.java:109)
> at com.fsck.k9.mail.store.imap.ImapFolder.copyMessages(ImapFolder.java:365)
> ... 10 more

no idea if it helps at anything

**Comment by Nihim** [comment]
another one, let me know if you don't need these.

```
K9-Mail version: 5.207
Device make: HUAWEI
Device model: FRD-L09
Android version: 7.0

com.fsck.k9.mail.MessagingException: Unable to get IMAP prefix
 at com.fsck.k9.mail.store.imap.ImapFolder.getPrefixedName(ImapFolder.java:93)
 at com.fsck.k9.mail.store.imap.ImapFolder.exists(ImapFolder.java:264)
 at com.fsck.k9.controller.MessagingController.verifyOrCreateRemoteSpecialFolder(MessagingController.java:1058)
 at com.fsck.k9.controller.MessagingController.synchronizeMailboxSynchronous(MessagingController.java:812)
 at com.fsck.k9.controller.MessagingController$36.run(MessagingController.java:4008)
 at com.fsck.k9.controller.MessagingController.runInBackground(MessagingController.java:204)
 at com.fsck.k9.controller.MessagingController.access$000(MessagingController.java:115)
 at com.fsck.k9.controller.MessagingController$1.run(MessagingController.java:173)
 at java.lang.Thread.run(Thread.java:776)
Caused by: java.io.IOException: readStringUntil(): end of stream reached
 at com.fsck.k9.mail.store.imap.ImapResponseParser.readStringUntil(ImapResponseParser.java:447)
 at com.fsck.k9.mail.store.imap.ImapResponseParser.parseTaggedResponse(ImapResponseParser.java:287)
 at com.fsck.k9.mail.store.imap.ImapResponseParser.readTaggedResponse(ImapResponseParser.java:75)
 at com.fsck.k9.mail.store.imap.ImapResponseParser.readResponse(ImapResponseParser.java:44)
 at com.fsck.k9.mail.store.imap.ImapResponseParser.readResponse(ImapResponseParser.java:30)
 at com.fsck.k9.mail.store.imap.ImapResponseParser.readStatusResponse(ImapResponseParser.java:88)
 at com.fsck.k9.mail.store.imap.ImapConnection.saslAuthPlain(ImapConnection.java:410)
 at com.fsck.k9.mail.store.imap.ImapConnection.saslAuthPlainWithLoginFallback(ImapConnection.java:385)
 at com.fsck.k9.mail.store.imap.ImapConnection.authenticate(ImapConnection.java:335)
 at com.fsck.k9.mail.store.imap.ImapConnection.open(ImapConnection.java:121)
 at com.fsck.k9.mail.store.imap.ImapFolder.getPrefixedName(ImapFolder.java:91)
 ... 8 more
```

**Comment by philipwhiuk** [comment]
These aren't useful. All it really means is you lost network connectivity. We log them mainly for debugging purposes - ideally we'd handle them somewhat better than a k9mail-errors folder but that's a fairly big chunk of work :(

**Comment by Nihim** [comment]
Alright, so will disable debug mode.

As far the fix goes, I 'll trust you with it, seeing I have no clue what's going on :)

Thanks for your work on K-9 !

**Comment by Nihim** [comment]
@cketti @philipwhiuk 

I did some simple tests. 

- I removed an account (A1) and re-added it. It stopped crashing and works properly.
- I exported the setting of another account (A2) and re-imported it. Again it works fine but I saw that it wasn't showing the syncing wheel on the notifications bar (only the above account A1 shows the syncing wheel). I removed again A2 and re-added it and now it works with the syncing wheel and all.

The rest of the accounts crash as normal.

So maybe it has (also) to do with the importing a backup from android 6 on android 7?
